luaui/Widgets/unit_smart_select.lua

```diff
@@ -68,7 +70,7 @@ for udid, udef in pairs(UnitDefs) do
 		ignoreUnits[udid] = true
 	end
 
-	local isMobile = (udef.canMove and udef.speed > 0.000001)  or  (includeNanosAsMobile and (udef.name == "armnanotc" or udef.name == "cornanotc"))
+	local isMobile = (udef.canMove and udef.speed > 0.000001)  or (includeNanosAsMobile and (udef.name == "armnanotc" or udef.name == "cornanotc"))
```

Comment:
since it's being touched, how about
```suggestion
	local isMobile = not udef.isImmobile or (includeNanosAsMobile and udef.isStaticBuilder)
```
and perhaps in the next line:
```
local builder = udef.isBuilder
```